packages/block-editor/src/components/block-list/block.js

```diff
@@ -372,13 +373,11 @@ function BlockListBlock( {
 		}
 	};
 
-	// Rendering the output
 	const isHovered = isBlockHovered && ! isPartOfMultiSelection;
 	const blockType = getBlockType( name );
+
 	// translators: %s: Type of block (i.e. Text, Image etc)
-	const blockLabel = sprintf( __( 'Block: %s' ), blockType.title );
-	// The block as rendered in the editor is composed of general block UI
-	// (mover, toolbar, wrapper) and the display of the block content.
+	const blockLabel = sprintf( __( 'Block: %s' ), __experimentalGetBlockLabel( blockType, attributes, ' - ' ) );
```

Comment:
@hypest @koke @Tug - it didn't take a lot of time to see this experimental method useful in the web context as well :)

Reply:
Hmm it is a different method though. The one used on mobile is `__experimentalGetAccessibilityLabel`, input parameters and the content returned is different as well. `__experimentalGetAccessibilityLabel` is for constructing a custom description for the block to be read by VoiceOver. It can be a long description too. For example it reads the caption and the alt text for the image block. AFAU `__experimentalGetBlockLabel` is used in places where a short description of block is needed.

Reply:
It would be great to have one method which would cover all those use cases through some sort of context. In your case it could be:
```js
getBlockLabel( clientId, { context: 'screen-reader` } );
```

Reply:
Sounds like a very good idea.

Reply:
Yep, they're a bit different at the moment. I'll have a look into it and see what the options are for consolidating the two.

Reply:
Did a quick audit of the labels produced by `__experimentalGetAccessibilityLabel ` for different blocks:
- Heading Block - outputs the heading level followed by the heading text e.g. "2. Introduction".
- Image Block - outputs the image `alt` concatenated with the `caption`, separated by a period. e.g. 'Mount Fuji. Mount Fuji in the spring'.
- Missing Block - outputs the original block title.
- More Block - outputs the custom text if there is some.
- Paragraph - outputs the paragraph text.

The Image block and Paragraph block descriptions could be pretty long for accessibility labels, but I think the others could work pretty well for both accessible and visual block labels.

When this function is used in the `BlockListBlock` component, some additional text is also concatenated, it'll produce something like 'Heading Block. Row 1. 2. Introduction' where the row number represents the position of the block.

Reply:
> The Image block and Paragraph block descriptions could be pretty long for accessibility labels, but I think the others could work pretty well for both accessible and visual block labels.

This is usually not a problem if there is no important information *after* that long text. So if block type, level, coordinates etc., all come first, and then the actual text is last, users can always interrupt speech once they know what that block is about. One could artificially limit that character count to 100 or 150, but that runs the potential of not being enough for some.

The friends at Gutenberg Mobile already do this block labeling in their navigation layer, at least for VoiceOver on iOS. Perhaps look at what they're doing and mirror that?

Reply:
I've consolidated the functionality now so that both mobile and web code uses `__experimentalGetAccessibilityLabel` and it should return the same format as the mobile function produced previously. I haven't tested it beyond unit tests for the function, though.

@pinarol Let me know what you think.

I think there's merit in what @gziolo proposes for visual labels. At the moment there's only the navigation link block that has a visual label, but others like the image, paragraph and heading block might require a slightly different label (or in some cases, no label at all).

Reply:
@MarcoZehe Thanks for the feedback! For some reason I didn't see your message until after the one I posted above.

I've kept the format of the text broadly the same across web and mobile, it follows the pattern established on mobile. Having said that the labels for each block themselves still need to be implemented separately on web as they're in different files. I plan to look at some of this next.

Reply:
@talldan this is looking good overall. But there's just one thing that sounds odd when VoiceOver is on and it should be fixed. If there's an unrecognized block the screen reader says 'Unrecognized block block', so there's an extra block at the end. I think it is related with the removal of this logic:

```
if ( name === 'core/missing' ) { // is the block unrecognized?	
			blockName = title;	
}
```
Missing blocks are already coming with 'Unrecognized Block' title. 

The rest looks good to me.

Reply:
Cool, thanks for spotting that @pinarol!